TypeSystem/flow-analysis/type_of_interest_A03_t05.dart

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that at type `T` is a type of interest for a variable `x`
+/// in a set of tested types `tested` if `tested` contains a type `S` such that
+/// `T` is `S`, or `T` is `NonNull(S)`.
+///
+/// @description Checks that a type `Never` cannot be made a type of interest
+/// for a variable.
+/// @author sgrekhov22@gmail.com
+
+import 'dart:async';
+
+test1() {
+  String s = "a";
+  if (s is Never) {}  // Try to make `Never` a type of interest
+  s = "b";
+  s.substring(0);
+}
+
+test2() {
+  String s = "a";
+  if (s is FutureOr<Never>) {}
```

Comment:
This would again require that `s` has a much more general declared type. `Object` should work.